src/backend/distributed/shared_library_init.c

```diff
@@ -664,3 +686,33 @@ NormalizeWorkerListPath(void)
 					PGC_S_OVERRIDE);
 	free(absoluteFileName);
 }
+
+
+static bool
+ReplicationFactorCheckHook(int *newval, void **extra, GucSource source)
+{
+	if (*newval != 1 && ReplicationModel == REPLICATION_MODEL_STREAMING)
+	{
+		GUC_check_errdetail("Cannot set replication factor above one when "
+							"\"citus.replication_model\" is \"streaming\".");
+
+		return false;
+	}
+
+	return true;
+}
+
+
+static bool
+ReplicationModelCheckHook(int *newval, void **extra, GucSource source)
+{
+	if (*newval == REPLICATION_MODEL_STREAMING && ShardReplicationFactor != 1)
+	{
+		GUC_check_errdetail("Cannot set replication model to \"streaming\" when "
+							"\"citus.shard_replication_factor\" greater than one.");
+
+		return false;
```

Comment:
> Depending on which value is checked first, you'll get different results

Can you give an example of the above? I think in this case it's impossible to get into a bad state. The only "order-dependent" thing going on is which error message is emitted.

To be fully clear, I think you're asking that we lazily check this only when the rep model is needed? That seems like a much worse user experience to me, as it far removes the effect of the setting from the first place we could reasonable detect that it is invalid, but if there are structural reasons we can't take the hook approach, fine by me I guess.